rasterimage/writer.go

```diff
@@ -0,0 +1,37 @@
+package rasterimage
+
+import (
+	"image/gif"
+	"image/jpeg"
+	"image/png"
+	"io"
+
+	"github.com/tdewolff/canvas"
+)
+
+// PNGWriter writes the canvas as a PNG file
+func PNGWriter(dpm float64) canvas.Writer {
```

Comment:
Initially it was a little unclear to me what the numeric parameter to these writers was (it is `dpm`), but I don't think we could clear that up (the same was present in the old design). Perhaps we could add a `DPI` and `DPM` function in the main package that creates a `Resolution` type (the same as `dpm float64`). Then people would write `PNGWriter(canvas.DPI(72.0))` which is clearer, any thoughts? (independent idea, this would not go in this PR)

Reply:
I have added a `PixelsPerMillimeter` type (called `density`, in reference to https://en.wikipedia.org/wiki/Pixel_density).

For easy conversion, I added a `const DPI = PixelsPerMillimeter(1 / 25.4)`

So that one can write: `PNGWriter(300*canvas.DPI)`

What do you think?